whittle/metrics/latency.py

```diff
@@ -0,0 +1,93 @@
+import torch
+import torch.profiler
+from torch.profiler import record_function
+
+
+# Define a sample language model (e.g., a simple RNN or transformer-based model)
+def get_total_cpu_gpu_runtime(prof):
+    # from torch.autograd.profiler_utils.py (_build_table)
+    # results is in micro-seconds
+    from torch.autograd import DeviceType
+
+    events = prof.events()
+    sum_self_cpu_time_total = sum([event.self_cpu_time_total for event in events])
+    sum_self_cuda_time_total = 0
+    for evt in events:
+        if evt.device_type == DeviceType.CPU:
+            # in legacy profiler, kernel info is stored in cpu events
+            if evt.is_legacy:
```

Comment:
Shouldn't we have another else block here if we don't use a legacy provider?

Reply:
I don't think there should be an else, the cpu time is already computed before the loop, the if is only for legacy events, where some times could be saved on cuda.

Reply:
ah right, makes sense